In [12]:
import pandas as pd
from zipfile import ZipFile
import yfinance as fp
import datetime

In [14]:
# with ZipFile("Data/NIFTY50_all.csv.zip","r") as ref:
#     unzip = ref.extractall("Data/Bnk_NIFTY50")
# ref.close()

In [3]:
today = datetime.date.today()
year  = today-datetime.timedelta(days=7300) #20 Years
Stocks = ['AAPL', 'MSFT', 'IBM', 'GE', 'JNJ']
data = fp.download(Stocks,start=year,end=today,group_by="ticker")
data

C:\Users\lipun\AppData\Local\Temp\ipykernel_17780\1640078772.py:4: FutureWarning: YF.download() has changed argument auto_adjust default to True
  data = fp.download(Stocks,start=year,end=today,group_by="ticker")
[*********************100%***********************]  5 of 5 completed


Ticker            MSFT                                                \
Price             Open        High         Low       Close    Volume   
Date                                                                   
2005-07-22   18.079355   18.322824   17.828928   17.863710  97558900   
2005-07-25   17.870661   18.016742   17.842835   17.870661  45174600   
2005-07-26   17.891535   17.905448   17.759367   17.766323  51476400   
2005-07-27   17.815015   17.947183   17.759365   17.891533  57977300   
2005-07-28   17.912395   17.981958   17.849789   17.912395  44738700   
...                ...         ...         ...         ...       ...   
2025-07-10  503.049988  504.440002  497.750000  501.480011  16492100   
2025-07-11  498.470001  505.029999  497.799988  503.320007  16459500   
2025-07-14  501.519989  503.970001  501.029999  503.019989  12058800   
2025-07-15  503.019989  508.299988  502.790009  505.820007  14927200   
2025-07-16  505.179993  506.720001  501.890015  505.619995  15146200   

Ticker             JNJ                                                ...  \
Price             Open        High         Low       Close    Volume  ...   
Date                                                                  ...   
2005-07-22   36.173404   36.173404   35.948724   36.128468   5398400  ...   
2005-07-25   36.089133   36.111602   35.555519   35.577988   6741900  ...   
2005-07-26   35.297146   35.594846   35.201656   35.201656  12640400  ...   
2005-07-27   35.443182   36.381218   35.398247   36.252029  10789500  ...   
2005-07-28   36.218336   36.420544   36.016124   36.358761   6391600  ...   
...                ...         ...         ...         ...       ...  ...   
2025-07-10  156.100006  159.240005  155.820007  157.690002   6732900  ...   
2025-07-11  157.089996  157.179993  155.179993  156.899994   7872000  ...   
2025-07-14  156.869995  157.470001  155.520004  156.820007  10185600  ...   
2025-07-15  156.360001  157.190002  154.800003  155.169998   6873200  ...   
2025-07-16  160.300003  166.119995  159.800003  164.779999  22130600  ...   

Ticker            AAPL                                                 \
Price             Open        High         Low       Close     Volume   
Date                                                                    
2005-07-22    1.305554    1.322384    1.304051    1.322384  301106400   
2005-07-25    1.322084    1.330800    1.314270    1.316674  294627200   
2005-07-26    1.322685    1.325690    1.303149    1.311264  268592800   
2005-07-27    1.317275    1.324489    1.282413    1.322084  283749200   
2005-07-28    1.317876    1.322385    1.301347    1.316374  251311200   
...                ...         ...         ...         ...        ...   
2025-07-10  210.509995  213.479996  210.029999  212.410004   44443600   
2025-07-11  210.570007  212.130005  209.860001  211.160004   39765800   
2025-07-14  209.929993  210.910004  207.539993  208.619995   38840100   
2025-07-15  209.220001  211.889999  208.919998  209.110001   42296300   
2025-07-16  210.300003  212.399994  208.639999  210.160004   47429600   

Ticker              GE                                                
Price             Open        High         Low       Close    Volume  
Date                                                                  
2005-07-22  102.833553  102.833553  101.753855  102.337479   3133979  
2005-07-25  102.716823  102.716823  101.257773  101.462036   3783503  
2005-07-26  101.637095  102.454164  101.228561  101.257744   3420410  
2005-07-27  101.286985  101.607971  101.111896  101.549614   3482466  
2005-07-28  101.783028  102.279102  101.549582  101.783028   3392115  
...                ...         ...         ...         ...       ...  
2025-07-10  250.610001  252.679993  246.750000  252.570007   5462500  
2025-07-11  251.500000  256.880005  251.399994  255.419998   7867200  
2025-07-14  256.190002  262.459991  256.000000  262.339996   7502200  
2025-07-15  263.920013  265.059998  2

In [4]:
new = []
for i in Stocks:
    df = data[i].copy()
    df = df.reset_index()
    df["Stocks"] = i
    new.append(df)
new_data = pd.concat(new)
new_data  

Price       Date        Open        High         Low       Close     Volume  \
0     2005-07-22    1.305554    1.322384    1.304051    1.322384  301106400   
1     2005-07-25    1.322084    1.330800    1.314270    1.316674  294627200   
2     2005-07-26    1.322685    1.325690    1.303149    1.311264  268592800   
3     2005-07-27    1.317275    1.324489    1.282413    1.322084  283749200   
4     2005-07-28    1.317876    1.322385    1.301347    1.316374  251311200   
...          ...         ...         ...         ...         ...        ...   
5022  2025-07-10  156.100006  159.240005  155.820007  157.690002    6732900   
5023  2025-07-11  157.089996  157.179993  155.179993  156.899994    7872000   
5024  2025-07-14  156.869995  157.470001  155.520004  156.820007   10185600   
5025  2025-07-15  156.360001  157.190002  154.800003  155.169998    6873200   
5026  2025-07-16  160.300003  166.119995  159.800003  164.779999   22130600   

Price Stocks  
0       AAPL  
1       AAPL  
2       AAPL  
3       AAPL  
4       AAPL  
...      ...  
5022     JNJ  
5023     JNJ  
5024     JNJ  
5025     JNJ  
5026     JNJ  

[25135 rows x 7 columns]

In [5]:
new_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 25135 entries, 0 to 5026
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   Date    25135 non-null  datetime64[ns]
 1   Open    25135 non-null  float64       
 2   High    25135 non-null  float64       
 3   Low     25135 non-null  float64       
 4   Close   25135 non-null  float64       
 5   Volume  25135 non-null  int64         
 6   Stocks  25135 non-null  object        
dtypes: datetime64[ns](1), float64(4), int64(1), object(1)
memory usage: 1.5+ MB


In [10]:
#Saving File
new_data.to_csv("Data/DataPreparation.csv",index=False)

<h1><b>FINAL REPORT: DATA-PREPARATION</b></h1>
    <li>
        Data-Source: <a href="yfinance">Yfinance</a>
    </li>
    <li>Converted Raw Dataset to Structured Format With a New Feature as Stocks</li>
    <li><b>Total Samples:</b> 25135</li>
    <li><b>Total Independent Features: </b>"Date","Open","High","Low" ,"Volume and "Stocks"</li>
    <li><b>Dependent Feature:</b> "Close"</li>
    <center><b><li>Stocks Names</li></b></center>
    <table>
        <th>Stocks</th>
        <th>Symbols</th>
        <tr><td>AAPL</td><td>Apple</td></tr>
        <tr><td>MSFT</td><td>Microsoft</td></tr>
        <tr><td>IBM</td><td>IBM</td></tr>
        <tr><td>GE</td><td>General Electric</td></tr>
       <tr><td>JNJ</td><td>Johnson & Johnson</td></tr>
    </table>
    <center><li><b>Feature Dtypes</b></li></center>
    <table>
        <th>Features</th>
        <th>DataTypes</th>
        <tr><td>Date</td><td>datetime64</td></tr>
        <tr><td>Open</td><td>float64</td></tr>
        <tr><td>High</td><td>float64</td></tr>
        <tr><td>Low</td><td>float64</td></tr>
        <tr><td>Close</td><td>float64</td></tr>
        <tr><td>Volume</td><td>int64</td></tr>
        <tr><td>Stocks</td><td>object</td></tr>
    </table>